src/libfetchers/git-utils.cc

```diff
@@ -726,7 +734,28 @@ struct GitSourceAccessor : SourceAccessor
             }
         }
 
-        return std::string((const char *) git_blob_rawcontent(blob.get()), git_blob_rawsize(blob.get()));
+        if (!state->applyFilters)
+            return std::string((const char *) git_blob_rawcontent(blob.get()), git_blob_rawsize(blob.get()));
+        else {
+            // Apply git filters including potential CRLF conversion
+            git_buf filtered = GIT_BUF_INIT;
+            git_blob_filter_options opts = GIT_BLOB_FILTER_OPTIONS_INIT;
+
+            opts.attr_commit_id = state->oid;
+            opts.flags = GIT_BLOB_FILTER_ATTRIBUTES_FROM_COMMIT;
```

Comment:
```suggestion
            opts.flags = GIT_BLOB_FILTER_ATTRIBUTES_FROM_COMMIT | GIT_BLOB_FILTER_NO_SYSTEM_ATTRIBUTES;
```

For reproducibility, we don't want the filtering to depend on the user's configuration.